### PR TITLE
Pull request for WAZO-76-multi-tenant

### PR DESCRIPTION
**Pull request by sileht**
Make it multi-tenant aware
fixup flake8 issues
Enable zuul
Add flake8 and integration target to tox

Depends-On: https://github.com/wazo-pbx/xivo-manage-db/pull/4
Depends-On: https://github.com/wazo-pbx/xivo-dao/pull/2
Depends-On: https://github.com/wazo-pbx/xivo-test-helpers/pull/9
Depends-On: https://github.com/wazo-pbx/xivo-lib-python/pull/21

### CODE REVIEW

**Comment by sileht** [comment]
recheck

**Comment by sileht** [comment]
recheck

**Comment by sileht** [comment]
recheck

**Comment by sileht** [comment]
Don't merge it yet, I haven't tested the xivo-upgrade part. But you can started reviewing it.

**Comment by sileht** [comment]
It's ready, I have also tested the xivo-upgrade side!

File: integration_tests/suite/helpers/database.py [hunk 1]
```diff
@@ -107,8 +136,6 @@ def insert_call_log_participant(self, **kwargs):
     def find_last_call_log(self):
         session = self.Session()
         call_log = session.query(CallLog).order_by(CallLog.date).first()
-        if call_log:
-            make_transient(call_log)
```
Comment (fblackburn1) [inline integration_tests/suite/helpers/database.py]: Can you extract your modification into a separate commit to explain why it was added and why you need to remove?

File: integration_tests/suite/helpers/database.py [hunk 2]
```diff
@@ -1,19 +1,48 @@
...
+    def _decorate(func):
+        @wraps(func)
+        def wrapped_function(self, *args, **kwargs):
+            with self.database.queries() as queries:
+                for call_log in call_logs:
+                    participants = call_log.pop('participants', [{
+                        'user_uuid': str(uuid.uuid4()),
+                        'tenant_uuid': VALID_TENANT,
+                        'role': 'source',
+                    }])
```
Comment (fblackburn1) [inline integration_tests/suite/helpers/database.py]: Why you need to populate one participants by default (from [previous state](https://github.com/wazo-pbx/wazo-call-logd/blob/master/integration_tests/suite/test_cdr.py#L43))? We try to keep [fixtures](https://github.com/wazo-pbx/wazo-chatd/blob/master/integration_tests/suite/helpers/fixtures/db.py) with the minimal parameters possible to exist. If the test need more, then it should define it explicitly

File: integration_tests/suite/test_cdr.py [hunk 3]
```diff
@@ -252,7 +307,12 @@ def test_given_no_call_logs_when_list_cdr_then_empty_list(self):
          'destination_name': '.tsèd',
          'direction': 'outbound',
          'source_exten': '7867',
-         'source_name': '.rùos'},
+         'source_name': '.rùos',
+         'participants': [{'user_uuid': USER_1_UUID,
+                           'tenant_uuid': VALID_TENANT,
+                           'line_id': '1',
+                           'tags': ['rh', 'Poudlard'],
+                           'role': 'destination'}]},
```
Comment (fblackburn1) [inline integration_tests/suite/test_cdr.py]: Why you modify the test to add a participant? A call log can exist without having a participant. If your code depend on participants, then you have a problem

File: integration_tests/suite/test_cdr.py [hunk 4]
```diff
@@ -730,15 +827,21 @@ def test_given_token_with_no_user_uuid_when_list_my_cdr_then_400(self):
...
+        {'date': '2017-04-11', 'participants': [{'user_uuid': USER_1_UUID,
+                                                 'tenant_uuid': VALID_TENANT}]},
+        {'date': '2017-04-12', 'participants': [{'user_uuid': USER_1_UUID,
+                                                 'tenant_uuid': VALID_TENANT}]},
+        {'date': '2017-04-13', 'participants': [{'user_uuid': USER_1_UUID,
+                                                 'tenant_uuid': VALID_TENANT}]},
+        {'date': '2017-04-14', 'participants': [{'user_uuid': USER_1_UUID,
+                                                 'tenant_uuid': VALID_TENANT}]},
+        {'date': '2017-04-15', 'participants': [{'user_uuid': USER_2_UUID,
+                                                 'tenant_uuid': VALID_TENANT}]},
```
Comment (fblackburn1) [inline integration_tests/suite/test_cdr.py]: If you do not care about the tenant_uuid in your test, then you can add the default tenant_uuid: VALID_TENANT in the fixture of call_log when a user_uuid is set. It will be less verbose and more readable

File: integration_tests/suite/helpers/database.py (hunk 1 above)
Comment (sileht) [reply integration_tests/suite/helpers/database.py]: Sure, once you make an alchemy object transient, lazy/sub_query relationships don't work anymore.

But it should be the reverse any usage of make_transient() should be commented as it's not recommended to use that method and often hide an other issue.

File: integration_tests/suite/helpers/database.py (hunk 2 above)
Comment (sileht) [reply integration_tests/suite/helpers/database.py]: Since you can only retrieve cdr that belong someone in REST API, all tests need at least one participants (except maybe one to show the cdr without participants doesn't show up in API).

File: integration_tests/suite/test_cdr.py (hunk 3 above)
Comment (sileht) [reply integration_tests/suite/test_cdr.py]: Yes it can exists, but since it's not belong your tenant you can't get it via API.

File: integration_tests/suite/test_cdr.py (hunk 4 above)
Comment (sileht) [reply integration_tests/suite/test_cdr.py]: I will do that.

File: integration_tests/suite/helpers/database.py (hunk 2 above)
Comment (sileht) [reply integration_tests/suite/helpers/database.py]: (just thinking out of loud) 

Or `a call_lod should also have a tenant_uuid, because all calls are from/to a tenant even if we don't known the participants yet`:
* if that false, the current implementation is ok
* if that true we need to find a how to link the `cel` with a `tenant_uuid` without using `participants`.

I haven't found this link with the information I have in a `cel`, do I miss something ?

File: integration_tests/suite/test_cdr.py (hunk 3 above)
Comment (fblackburn1) [reply integration_tests/suite/test_cdr.py]: Yes we MUST get it via API. Probably linked to the master tenant?

File: integration_tests/suite/helpers/database.py (hunk 2 above)
Comment (fblackburn1) [reply integration_tests/suite/helpers/database.py]: For me all cel should have a tenant. With or without participants. In cel we have the context that you can use to link with a tenant

File: integration_tests/suite/helpers/database.py (hunk 1 above)
Comment (fblackburn1) [reply integration_tests/suite/helpers/database.py]: Agreed that usage of make_transient should be commented or not used.
In this case, I'm just surprised that you can remove it without side effect (or you fix something that I did'nt see?)

I'm not sure to understand the difference between `with self.Session()` instead `session.commit()`. Have you some documentation about the Session context manager?

Comment (sileht) [reply integration_tests/suite/helpers/database.py]: > Agreed that usage of make_transient should be commented or not used.
> In this case, I'm just surprised that you can remove it without side effect (or you fix something that I didn't see?)

I have ran test in separate commit without issue, so...

> I'm not sure to understand the difference between `with self.Session()` instead `session.commit()`. Have you some documentation about the Session context manager?

I was thinking of these:
* https://github.com/wazo-pbx/wazo-webhookd/blob/master/wazo_webhookd/plugins/subscription/service.py#L41
* https://github.com/wazo-pbx/wazo-call-logd/blob/master/wazo_call_logd/database/dao.py#L47

File: integration_tests/suite/test_cdr.py (hunk 3 above)
Comment (sileht) [reply integration_tests/suite/test_cdr.py]: We have found ton of contexts and associated tenant with them.

File: wazo_call_logd/plugins/tenant_migration/http.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+    def __init__(self, service):
+        self._service = service
+
+    @required_acl('webhookd.tenant-upgrade')
+    def post(self):
+        for item in request.json['users']:
+            self._service.update_participants(**item)
+        for item in request.json['contexts']:
+            self._service.update_contexts(**item)
```
Comment (fblackburn1) [inline wazo_call_logd/plugins/tenant_migration/http.py]: We should add a step to remove all non migrated tenant. By deleting the call log and the call_log_id (in CEL table), the cron will automatically regenerate these entries.

File: wazo_call_logd/database/dao.py [hunk 6]
```diff
@@ -58,11 +58,14 @@ def new_session(self):
...
         with self.new_session() as session:
-            cdr = session.query(CallLogSchema).options(joinedload('participants'),
-                                                       subqueryload('source_participant'),
-                                                       subqueryload('destination_participant')).get(cdr_id)
+            query = session.query(CallLogSchema).options(joinedload('participants'),
+                                                         subqueryload('source_participant'),
+                                                         subqueryload('destination_participant'))
+            query = self._apply_filters(query, {'tenant_uuids': tenant_uuids})
+            query = query.filter(CallLogSchema.id == cdr_id)
+            cdr = query.one_or_none()
```
Comment (fblackburn1) [inline wazo_call_logd/database/dao.py]: By curiosity, since `cdr_ir` is the primary_key, why not continue to use `.get(cdr_id)` instead of `filter` + `one_or_none`

File: wazo_call_logd/database/dao.py [hunk 7]
```diff
@@ -149,6 +157,17 @@ def _apply_filters(self, query, params):
...
 
+        if params.get('tenant_uuids'):
+            filters = [CallLogSchema.participant_tenant_uuids.contains(str(tenant_uuid))
+                       for tenant_uuid in params['tenant_uuids']]
+            filters.extend([
+                CallLogSchema.requested_tenant_uuid.in_(params['tenant_uuids']),
+                CallLogSchema.requested_internal_tenant_uuid.in_(params['tenant_uuids']),
+                CallLogSchema.source_internal_tenant_uuid.in_(params['tenant_uuids']),
+                CallLogSchema.destination_internal_tenant_uuid.in_(params['tenant_uuids']),
+            ])
```
Comment (fblackburn1) [inline wazo_call_logd/database/dao.py]: As mentioned above, I think we should only search through the `tenant_uuid` of the call log (not participant)

File: wazo_call_logd/generator.py [hunk 8]
```diff
@@ -54,3 +57,16 @@ def _get_interpretor(self, cels):
                 return interpretor
 
         raise RuntimeError('Could not find suitable interpretor in {}'.format(self._cel_interpretors))
+
+    @staticmethod
+    def _tenant_checker(call_log):
+        for prefix in ('requested', 'requested_internal',
+                       'source_internal', 'destination_internal'):
+            if getattr(call_log, '%s_tenant_uuid' % prefix):
```
Comment (fblackburn1) [inline wazo_call_logd/generator.py]: use `format`
I must admit that I don't really like to build the attribute name dynamically, it can be painful when doing `grep`. Have you an opinion? or my argument is invalid? :stuck_out_tongue:

File: wazo_call_logd/plugins/cdr/resource.py [hunk 9]
```diff
@@ -84,62 +94,64 @@ def _output_csv(data, code, http_headers=None):
     return response
 
 
-class CDRResource(AuthResource):
-
+class CDRAuthResource(AuthResource):
     representations = {'text/csv; charset=utf-8': _output_csv}
 
     def __init__(self, cdr_service):
+        super().__init__()
```
Comment (fblackburn1) [inline wazo_call_logd/plugins/cdr/resource.py]: Only for styling/best practices?

File: wazo_call_logd/plugins/cdr/resource.py [hunk 10]
```diff
@@ -84,62 +94,64 @@ def _output_csv(data, code, http_headers=None):
...
         return CDRSchemaList().dump(cdrs).data
 
 
-class CDRUserMeResource(AuthResource):
-
-    representations = {'text/csv; charset=utf-8': _output_csv}
+class CDRUserMeResource(CDRAuthResource):
 
     def __init__(self, auth_client, cdr_service):
+        super(CDRUserMeResource, self).__init__(cdr_service)
```
Comment (fblackburn1) [inline wazo_call_logd/plugins/cdr/resource.py]: You use two syntax of `super()` 
1- `super().__init__()`
2- `super(CDRUserMeResource, self).__init__(cdr_service)`
I propose to only use the `super()` without the class name and `self`

File: wazo_call_logd/plugins/cdr/resource.py [hunk 11]
```diff
@@ -84,62 +94,64 @@ def _output_csv(data, code, http_headers=None):
...
         self.auth_client = auth_client
-        self.cdr_service = cdr_service
 
     @required_acl('call-logd.users.me.cdr.read')
     def get(self):
         args = CDRListRequestSchema().load(request.args).data
         user_uuid = get_token_user_uuid_from_request(self.auth_client)
         args['me_user_uuid'] = user_uuid
+        args['tenant_uuids'] = [Tenant.autodetect().uuid]
```
Comment (fblackburn1) [inline wazo_call_logd/plugins/cdr/resource.py]: I don't think we should use `Tenant.autodetect()` for the the `/users/me/`, since the name `autodetect` is to refer from header or from token. We should use `Token.tenant_uuid`

File: integration_tests/suite/test_tenant_migration.py [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+            ],
+        }
+        headers = {'X-Auth-Token': MASTER_TOKEN, 'Content-Type': 'application/json'}
+        resp = requests.post(url, json=payload, headers=headers, verify=False)
+        resp.raise_for_status()
+
+        with self.database.queries() as queries:
+            with closing(queries.Session()) as session:
+                query = session.query(CallLogParticipant)
+                for clp in query.all():
```
Comment (fblackburn1) [inline integration_tests/suite/test_tenant_migration.py]: Avoid to use acronym when it is not inline code. `call_log_participant` or `participant`

File: integration_tests/suite/test_tenant_migration.py [hunk 13]
```diff
@@ -0,0 +1,113 @@
...
+                    if clp.user_uuid in [USER_1_UUID, USER_2_UUID]:
+                        assert_that(clp.tenant_uuid, USERS_TENANT)
+                    else:
+                        assert_that(clp.tenant_uuid, OTHER_TENANT)
+
+                query = session.query(CallLog)
+                for cl in query.all():
```
Comment (fblackburn1) [inline integration_tests/suite/test_tenant_migration.py]: be explicit: `call_log` instead of `cl`

File: integration_tests/suite/test_tenant_migration.py [hunk 14]
```diff
@@ -0,0 +1,113 @@
...
+                    for prefix in ('requested', 'requested_internal',
+                                   'source_internal', 'destination_internal'):
+                        context = getattr(cl, '%s_context' % prefix)
+                        tenant_uuid = getattr(cl, '%s_tenant_uuid' % prefix)
```
[the remark this answers is not in the document]
Comment (sileht) [reply integration_tests/suite/test_tenant_migration.py]: Eheh, I have such a bad habit on this, openstack always wanted to stay in the past on this.

[the remark this answers is not in the document]
Comment (sileht) [reply integration_tests/suite/test_tenant_migration.py]: We may need to add this: https://pypi.org/project/flake8-string-format/

File: integration_tests/suite/test_cdr.py [hunk 15]
```diff
@@ -214,10 +202,70 @@ def test_given_id_when_get_cdr_by_id_csv_then_get_cdr_by_id_csv(self):
...
+    ])
+    def test_get_cdr_by_id_multitenant(self):
+        self.call_logd.set_token(USER_1_TOKEN)
+        result = self.call_logd.cdr.get_by_id(10)
+        assert_that(result, has_entries(source_user_uuid=USER_1_UUID,
+                                        source_tenant_uuid=USERS_TENANT))
+
+        result = self.call_logd.cdr.get_by_id(11)
+        assert_that(result, has_entries(source_user_uuid=USER_2_UUID,
+                                        source_tenant_uuid=USERS_TENANT))
```
[the remark this answers is not in the document]
Comment (sileht) [reply integration_tests/suite/test_cdr.py]: Here I check that `USER_1` (with enough permissions) can see `USER_1` and  `USER_2` cdr and not `OTHER_USER` cdr

File: integration_tests/suite/test_cdr.py [hunk 16]
```diff
@@ -214,10 +202,70 @@ def test_given_id_when_get_cdr_by_id_csv_then_get_cdr_by_id_csv(self):
...
+                    message=contains_string_ignoring_case('no cdr found'),
+                    details=has_key('cdr_id')
+                )
+            )
+        )
+
+        self.call_logd.set_token(OTHER_USER_TOKEN)
+        result = self.call_logd.cdr.get_by_id(12)
+        assert_that(result, has_entries(source_user_uuid=OTHER_USER_UUID,
+                                        source_tenant_uuid=OTHER_TENANT))
```
[the remark this answers is not in the document]
Comment (sileht) [reply integration_tests/suite/test_cdr.py]: I first check to get cdr `12` with `USER_1_TOKEN` it returns 404, it could mean, it really doesn't exists or it exists but its not belong my tenant. So I check with OTHER_USER_TOKEN to ensure its the seconds hypotheses.

File: integration_tests/suite/test_cdr.py [hunk 17]
```diff
@@ -214,10 +202,70 @@ def test_given_id_when_get_cdr_by_id_csv_then_get_cdr_by_id_csv(self):
...
+
+        self.call_logd.set_token(MASTER_TOKEN)
+        result = self.call_logd.cdr.get_by_id(10)
+        assert_that(result, has_entries(source_user_uuid=USER_1_UUID,
+                                        source_tenant_uuid=USERS_TENANT))
+
+        result = self.call_logd.cdr.get_by_id(11)
+        assert_that(result, has_entries(source_user_uuid=USER_2_UUID,
+                                        source_tenant_uuid=USERS_TENANT))
+
```
[the remark this answers is not in the document]
Comment (sileht) [reply integration_tests/suite/test_cdr.py]: I just wanted I be sure I see all CDRs, but yeah, the tested code path is the same as ID 10

File: integration_tests/suite/test_cdr.py [hunk 18]
```diff
@@ -214,10 +202,70 @@ def test_given_id_when_get_cdr_by_id_csv_then_get_cdr_by_id_csv(self):
...
+        result = self.call_logd.cdr.get_by_id(12)
+        assert_that(result, has_entries(source_user_uuid=OTHER_USER_UUID,
+                                        source_tenant_uuid=OTHER_TENANT))
+
```
[the remark this answers is not in the document]
Comment (sileht) [reply integration_tests/suite/test_cdr.py]: This one is different, I check MASTER_TOKEN can see cdr of two other child tenant USERS_TENANT and OTHER_TENANT, not just one.

File: wazo_call_logd/plugins/cdr/resource.py (hunk 10 above)
Comment (sileht) [reply wazo_call_logd/plugins/cdr/resource.py]: +1 for 2), I wrote too many code that need to support 2 :), But I'm really happy to do py3 only syntax !

File: wazo_call_logd/plugins/cdr/resource.py (hunk 11 above)
Comment (sileht) [reply wazo_call_logd/plugins/cdr/resource.py]: Changed, since user_uuid is unique and come from tenant, the user will just see nothing.

File: integration_tests/suite/test_tenant_migration.py (hunk 12 above)
Comment (sileht) [reply integration_tests/suite/test_tenant_migration.py]: done

File: integration_tests/suite/test_tenant_migration.py (hunk 13 above)
Comment (sileht) [reply integration_tests/suite/test_tenant_migration.py]: done

File: wazo_call_logd/plugins/cdr/resource.py (hunk 9 above)
Comment (sileht) [reply wazo_call_logd/plugins/cdr/resource.py]: Just best practices, I don't want to known if `AuthResource` have `__init__` implemented or not. Maybe today it haven't, who known later. Python always call the object() one for us, but it's only magisk, in some other oom languages it doesn't.

When there is too much inheritances, you don't want to check all parents, grand parents, ... class to see if someone in the tree have a custom __init__. Always adding it always work, while missing one can introduce nasty bug (like code looks like to work, but one var is not initialized as expected).

File: integration_tests/suite/test_tenant_migration.py (hunk 14 above)
[the remark this answers is not in the document]
Comment (fblackburn1) [reply integration_tests/suite/test_tenant_migration.py]: This plugin does only check syntax of `format`. If we do not use `format` then it doesn't raise error :(

File: integration_tests/suite/test_cdr.py (hunk 18 above)
[the remark this answers is not in the document]
Comment (fblackburn1) [reply integration_tests/suite/test_cdr.py]: Ok I see, but the code path is also the same. If the MASTER_TENANT can see the USERS_TENANT, then there are no reason why it cannot see OTHER_TENANT. In fact, USERS_TENANT is the same thing as OTHER_TENANT, e.i. a sub tenant of MASTER_TENANT.
I don't have a big objection to keep it, just I don't want to think about why we test it twice.

File: integration_tests/suite/test_cdr.py (hunk 17 above)
[the remark this answers is not in the document]
Comment (fblackburn1) [reply integration_tests/suite/test_cdr.py]: same as below: I don't have a big objection to keep it, just I don't want to think about why we test it twice.

File: integration_tests/suite/test_cdr.py (hunk 15 above)
[the remark this answers is not in the document]
Comment (fblackburn1) [reply integration_tests/suite/test_cdr.py]: I must admit that I still don't see the value to test ID 11 :sweat_smile:

File: integration_tests/suite/test_cdr.py (hunk 16 above)
[the remark this answers is not in the document]
Comment (fblackburn1) [reply integration_tests/suite/test_cdr.py]: Is the only one that I can understand. But I have to say that we must trust our precondition (otherwise the test is invalid), so we must trust that call log exists

File: wazo_call_logd/plugins/tenant_migration/http.py (hunk 5 above)
Comment (fblackburn1) [reply wazo_call_logd/plugins/tenant_migration/http.py]: Well it can also cause some trouble if the resource don't exist anymore. participants information and context can be not fuond... so is maybe not a good idea :/

File: integration_tests/suite/test_tenant_migration.py (hunk 14 above)
[the remark this answers is not in the document]
Comment (sileht) [reply integration_tests/suite/test_tenant_migration.py]: > This plugin does only check syntax of `format`. If we do not use `format` then it doesn't raise error :(

sniff

File: integration_tests/suite/test_cdr.py (hunk 15 above)
[the remark this answers is not in the document]
Comment (sileht) [reply integration_tests/suite/test_cdr.py]: Even the code path is the same as USER_1, from user point of view I want to check USER_1 can see other USER_1 of the TENANT and nothing else.

(Just an impossible example): Imagine I wrote bad code that makes the final sql query having "user_uuid == <USER_UUID_OF_CURRENT_TOKEN>", if I haven't this test I may not see it.

In general, I add all tests that seems logic to check the users point view of the feature, even if two assertions check the same code path.

tests should not take care of how it's implemented. It just, sends input to something, to assert the output with valid hypotheses/scenario.

File: wazo_call_logd/database/dao.py (hunk 6 above)
Comment (sileht) [reply wazo_call_logd/database/dao.py]: get() will also returns call logs that not from my allowed tenants.

File: integration_tests/suite/test_cdr.py [hunk 19]
```diff
@@ -622,16 +677,18 @@ def test_given_call_logs_when_list_cdr_with_user_uuid_then_list_matching_cdr(sel
...
         self.call_logd.set_token(SOME_TOKEN)
+
         results = self.call_logd.cdr.list_from_user(user_uuid=USER_3_UUID)
 
         assert_that(
             results,
             has_entries(
                 filtered=1,
-                total=3,
+                total=2,
```
[the remark this answers is not in the document]
Comment (fblackburn1) [reply integration_tests/suite/test_cdr.py]: related to https://github.com/wazo-pbx/wazo-call-logd/pull/9

[the remark this answers is not in the document]
Comment (sileht) [reply integration_tests/suite/test_cdr.py]: thanks for answering for me :)

**Comment by sileht** [comment]
recheck

File: wazo_call_logd/generator.py [hunk 20]
```diff
@@ -16,8 +16,13 @@
 
 class CallLogsGenerator(object):
 
-    def __init__(self, cel_interpretors):
+    def __init__(self, confd, cel_interpretors):
+        self.confd = confd
         self._cel_interpretors = cel_interpretors
+        self._service_tenant_uuid = None
+
+    def set_token(self, token):
```
Comment (fblackburn1) [inline wazo_call_logd/generator.py]:
```suggestion
    def set_default_tenant(self, token):
```
or something more explicit `set_default_fallback_tenant_uuid_from_service_call_logd_token_details_:trollface:()`

File: wazo_call_logd/plugins/tenant_migration/service.py [hunk 21]
```diff
@@ -0,0 +1,61 @@
...
+        try:
+            yield session
+            session.commit()
+        except BaseException:
+            session.rollback()
+            raise
+        finally:
+            self._Session.remove()
+
+    def set_token(self, token):
```
Comment (fblackburn1) [inline wazo_call_logd/plugins/tenant_migration/service.py]: I don't know if the name should reflect what the method does or where it should be used ?

File: wazo_call_logd/plugins/tenant_migration/service.py [hunk 22]
```diff
@@ -0,0 +1,61 @@
...
+        self._service_tenant_uuid = token["metadata"]["tenant_uuid"]
+
+    def update_contexts(self, context, tenant_uuid):
+        with self.rw_session() as session:
+            for field in CONTEXT_ATTRIBUTES:
```
Comment (fblackburn1) [inline wazo_call_logd/plugins/tenant_migration/service.py]: Why need to loop on a value that is not used and make many time the same request?

File: wazo_call_logd/cel_interpretor.py [hunk 23]
```diff
@@ -63,7 +65,10 @@ def find_main_internal_extension(confd, channame):
         return
 
     main_extension = extensions[0]
-    logger.debug('Found main internal extension %s@%s', main_extension['exten'], main_extension['context'])
+    logger.debug('Found main internal extension %s@%s (%s)',
+                 main_extension['exten'],
+                 main_extension['context'],
+                 main_extension['tenant_uuid'])
```
Comment (fblackburn1) [inline wazo_call_logd/cel_interpretor.py]: confd doesn't return `tenant_uuid` for association line_extension

File: wazo_call_logd/cel_interpretor.py [hunk 24]
```diff
@@ -63,7 +65,10 @@ def find_main_internal_extension(confd, channame):
         return
 
     main_extension = extensions[0]
```
Comment (fblackburn1) [inline wazo_call_logd/cel_interpretor.py]:
```suggestion
    main_extension['tenant_uuid'] = line['tenant_uuid']
```

File: wazo_call_logd/generator.py (hunk 20 above)
Comment (sileht) [reply wazo_call_logd/generator.py]: I used set_default_tenant_uuid(()

File: wazo_call_logd/plugins/tenant_migration/service.py (hunk 21 above)
Comment (sileht) [reply wazo_call_logd/plugins/tenant_migration/service.py]: same here

File: wazo_call_logd/plugins/tenant_migration/service.py (hunk 22 above)
Comment (sileht) [reply wazo_call_logd/plugins/tenant_migration/service.py]: Thx, it's a leftover when I merged all *_tenant_uuid to only one.

File: wazo_call_logd/cel_interpretor.py (hunk 24 above)
Comment (sileht) [reply wazo_call_logd/cel_interpretor.py]: done!

**Comment by sileht** [comment]
recheck

**Comment by sileht** [comment]
recheck

**Comment by sileht** [comment]
recheck

**Comment by sileht** [comment]
recheck

**Comment by sileht** [comment]
regate

**Comment by sileht** [comment]
rabbitmq container have failed to start for one test...
